system-contracts/contracts/precompiles/EcAdd.yul

```diff
@@ -291,150 +65,27 @@ object "EcAdd" {
             let x2 := calldataload(64)
             let y2 := calldataload(96)
```

Comment:
You copy `x1`/`y1`/`x2`/`y2` into stack variables and then store them in memory (`mstore`). Instead, you can do `calldatacopy`, that would be less code and probably cheaper